### DISCUSSION TITLE
Issue Terraform on OKD 4.9 with baremetal

### DISCUSSION THREAD

**Discussion by JeanNNI** [okd-project/okd#1704] [Q&A]
Hi, i tried install OKD with baremetal but i have an error when put "openshift install cluster"
./openshift-install create cluster. I have install terraform and libvirtd

This log,

INFO Creating infrastructure resources...
ERROR Missing required providers.
ERROR The following provider constraints are not met by the currently-installed
ERROR provider plugins:

ERROR * ironic (any version)
ERROR * libvirt (any version)
ERROR Terraform can automatically download and install plugins to meet the given
ERROR constraints, but this step was skipped due to the use of -get-plugins=false
ERROR and/or -plugin-dir on the command line.

ERROR If automatic installation is not possible or desirable in your environment,
ERROR you may manually install plugins by downloading a suitable distribution package
ERROR and placing the plugin's executable file in one of the directories given in
ERROR by -plugin-dir on the command line, or in the following directory if custom
ERROR plugin directories are not set:
ERROR     terraform.d/plugins/linux_amd64

ERROR Error: missing provider "libvirt"
ERROR Error: missing provider "ironic"
FATAL failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed to initialize Terraform

**Comment by LorbusChris** [comment]
Please try to install a more recent release.

**Reply by LorbusChris** [reply]
Also, there is no need to install terraform manually. It's part of the installer binary

**Reply by JeanNNI** [reply]
Hi, well i tried "upgrade" version OKD but occurs similar situation

? Platform baremetal
? Provisioning Network Disabled
? External bridge ens160
? Add a Host: control plane
? Name master
? BMC Address ipmi://192.168.80.132
? BMC Username master
? BMC Password *********
X Sorry, your reply was invalid: invalid character 'F' looking for beginning of value
? Pull Secret [? for help] 
FATAL failed to fetch Metadata: failed to fetch dependency of "Metadata": failed to fetch dependency of "Cluster ID": failed to generate asset "Install Config": 
invalid install config: [platform.baremetal.apiVIPs: Invalid value: "15.197.X.X": VIP for API must not be one of the Ingress VIPs, platform.baremetal.apiVIPs:
 Invalid value: "15.197.x.x": IP expected to be in one of the machine networks: 10.0.0.0/16, platform.baremetal.ingressVIPs: Invalid value: "15.197.x.x":
  IP expected to be in one of the machine networks: 10.0.0.0/16, platform.baremetal.Hosts: Required value: not enough hosts found (1) to support all the configured ControlPlane replicas (3)]

I need only Controplane 1 without workes, therefore i created "install-config.yaml" and i run this comand "openshift-baremetal-install create install-config" and "openshift-baremetal-install create cluster" and log its similar situation shown above
